dbt_copilot_helper/templates/pipelines/codebase/overrides/cfn.patches.yml

```diff
@@ -0,0 +1,239 @@
+          - Name: AWS_ACCOUNT_ID
+            Value: !Sub '${AWS::AccountId}'
+          - Name: PARTITION
+            Value: !Ref AWS::Partition
+      ServiceRole: !GetAtt BuildProjectRole.Arn
+      Source:
+        Auth:
+          Type: OAUTH
+        GitCloneDepth: 0
+        GitSubmodulesConfig:
+          FetchSubmodules: true
+        Location: https://github.com/{{ codebase.repository }}.git
+        Type: GITHUB
+        BuildSpec: |
+          version: 0.2
+          env:
+            parameter-store:
+              SLACK_CHANNEL_ID: "/notifications/slack_channel_id"
+              SLACK_TOKEN: "/notifications/slack_oauth_token"
+          phases:
+            install:
+              commands:
+                - |
+                  if [ -f .copilot/phases/install.sh ]; then
+                    bash .copilot/phases/install.sh
+                  fi
+            pre_build:
+              commands:
+                - |
+                  if [ -f .copilot/phases/pre_build.sh ]; then
+                    bash .copilot/phases/pre_build.sh
+                  fi
+            build:
+              commands:
+                - |
+                  if [ -f .copilot/phases/build.sh ]; then
+                    bash .copilot/phases/build.sh
+                  fi
+                - /work/cli build --publish
+            post_build:
+              commands:
+                - |
+                  if [ -f .copilot/phases/post_build.sh ]; then
+                    bash .copilot/phases/post_build.sh
+                  fi
+      TimeoutInMinutes: 30
+      Triggers:
+        BuildType: BUILD
+        FilterGroups:
+          - - Type: EVENT
+              Pattern: PUSH
+            - Type: HEAD_REF
+              Pattern: ^refs/heads/{{ codebase.branch }}$
+        Webhook: true
+      Visibility: PRIVATE
+
+# Create an ECR repository for application images
+- op: add
+  path: /Resources/ECRRepository
+  value:
+    Type: AWS::ECR::Repository
+    Properties:
+      RepositoryName: {{ app_name }}/{{ codebase.name }}
+      EmptyOnDelete: true
+      ImageScanningConfiguration:
+        ScanOnPush: true
+      ImageTagMutability: MUTABLE
+
+#  Allow the pipeline to describe ECR repository images
+- op: add
+  path: /Resources/PipelineRolePolicy/Properties/PolicyDocument/Statement/0/Action/-
+  value: ecr:DescribeImages
+
+# Replace the source code action with a trigger on push to ECR
+- op: replace
+  path: /Resources/Pipeline/Properties/Stages/0/Actions/0
+  value:
+    Name: ImagePublished
+    ActionTypeId:
+      Category: Source
+      Owner: AWS
+      Version: 1
+      Provider: ECR
+    Configuration:
+      RepositoryName: {{ app_name }}/{{ codebase.name }}
+    OutputArtifacts:
+      - Name: ECRMetadata
+    RunOrder: 1
+
+# Remove the codebuild step from the pipeline
+- op: remove
+  path: /Resources/Pipeline/Properties/Stages/1
+
+# Environments
+{% for environment in codebase.environments %}
+# {{ environment.name }}
```

Comment:
```suggestion
```

Reply:
Wanted to leave these in as comments to quickly indicate to users which environment this step relates to.

Reply:
That's what I understood from this as well. This will be rendered as a comment with the environment name, as seen in this file: [tests/copilot_helper/expected_files/pipeline/pipelines/application/overrides/cfn.patches.yml](https://github.com/uktrade/copilot-tools/pull/216/files#diff-2865558b4f968dded04ea50e24d7fe846e5396cd70e25b8db25123706ea3094c)

Reply:
Oh, yeah, I forgot it was in a template, I just saw a redundant looking comment. All good.